cmd/dep/status.go

```diff
@@ -461,7 +463,41 @@ func formatVersion(v gps.Version) string {
 	return v.String()
 }
 
-func collectConstraints(ptree pkgtree.PackageTree, p *dep.Project, sm gps.SourceManager) map[string][]gps.Constraint {
-	// TODO
-	return map[string][]gps.Constraint{}
+// collectConstraints collects constraints declared by all the dependencies.
+func collectConstraints(ctx *dep.Ctx, ptree pkgtree.PackageTree, p *dep.Project, sm gps.SourceManager) map[string][]gps.Constraint {
+	constraintCollection := make(map[string][]gps.Constraint)
+
+	// Get direct deps of the root project
+	_, directDeps, err := getDirectDependencies(sm, p)
+	if err != nil {
+		ctx.Err.Println("Error getting direct deps:", err)
+	}
+	// Create a root analyzer
+	rootAnalyzer := newRootAnalyzer(false, ctx, directDeps, sm)
+
+	// Iterate through the locked projects and collect constrains of all the projects
+	for _, proj := range p.Lock.Projects() {
+		manifest, _, err := sm.GetManifestAndLock(proj.Ident(), proj.Version(), rootAnalyzer)
+		if err != nil {
+			ctx.Err.Println("Error getting manifest and lock:", err)
+		}
+
+		// Get project constraints
+		pc := manifest.DependencyConstraints()
+
+		// Iterate through the project constraints to get individual dependency
+		// project and constraint values.
+		for pr, pp := range pc {
+			// Check if a collection exists for a project root and append to
+			// existing one or create a new constraint collection.
+			if cc, exists := constraintCollection[string(pr)]; exists {
```

Comment:
i think this can be safely shortened to a one-liner, as lookups to nonexistent keys on maps will return the zero value for the map's value type:

```go
constraintCollection[string(pr)] = append(constraintCollection[string(pr)], pp.Constraint)
```